Fix #5789: Pin Certs with Chromium API

Brandon-T opened the pull request:
## Security Review
- https://github.com/brave/security/issues/1127

## Summary of Changes
- Pin Certs with Chromium API
- Display appropriate error pages when pinning fails or certificate validation fails.

<!-- Enter a ticket number for this PR, create a new one if it is not there yet. -->
This pull request fixes #5789

## Screenshots
![image](https://user-images.githubusercontent.com/1530031/212423358-3ca233f9-dcae-4127-96cc-8f63472b1be5.png)


## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes


## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

stoletheminerals reviewed (changes requested):
We should not allow a user to dismiss the error and proceed

<img width="657" alt="image" src="https://user-images.githubusercontent.com/10810135/212725422-915df31e-bc8f-4b12-8247-dfe5432e80b1.png">

Brandon-T committed:
Pin Certs with Chromium API and Display appropriate error pages when pinning fails or certificate validation fails.
Add Unit Test for Pinning Certificates.
Update signature to pass in the port which is required when validating against the Pinsets.

Switch to async-await API

Changed error handling to let the system handle the navigation otherwise the navigation stack is broken
